File: frontend/types/notificacao.ts
export type Notificacao = {
    id: number;
    usuario_id: number;
    tipo_notificacao:
        | "avaliacao"
        | "comentario"
        | "seguidor"
        | "mensagem"
        | "voto_avaliacao"
        | "denuncia_resolvida"
        | "advertencia"
        | "conta_inativada";
    titulo: string;
    conteudo: string;
    referencia_id?: number;
    lida: boolean;
    criado_em: string;
};

File: backend/src/modelos/Notificacao.go
package modelos

import "time"

type Notificacao struct {
	ID              uint64    `json:"id"`
	UsuarioID       uint64    `json:"usuario_id"`
	TipoNotificacao string    `json:"tipo_notificacao"`
	Titulo          string    `json:"titulo"`
	Conteudo        string    `json:"conteudo"`
	ReferenciaID    *uint64   `json:"referencia_id,omitempty"`
	Lida            bool      `json:"lida"`
	CriadoEm        time.Time `json:"criado_em"`
}
